src/quail/SensorTask.cpp

```diff
@@ -0,0 +1,41 @@
+#include "PressureSensor.hpp"
+#include "LoadSensor.hpp"
+#include "ThermalSensor.hpp"
+
+#include "SensorTask.hpp"
+
+SensorTask::SensorTask(uint8_t priority) : Task(priority, "Sensor"){};
+
+Sensor SensorTask::sensors[NUM_SENSORS] = 
+{
+    PressureSensor(0, "PT1", Ad7124::AIN1Input,PressureSensor::RANGE_1000) // channel, ADC in port, PT range
+};
+
+void SensorTask::activity()
+{
+    while(true)
+    {
+        adcdata_t adcpacket;
+        adcbuf.receive(adcpacket, NEVER); // wait until an adc packet is ready, state goes BLOCKED until not empty
+        Sensor cur_sensor = sensors[adcpacket.channel]; // sensor for which we just received data
```

Comment:
Is this a copy opperator?

Reply:
As is, its a copy - but I've actually changed this in the commit I'm about to push, because of issues with typecasting in the Sensors array. Instead, I'll just grab a pointer to the appropriate sensor object (which allows me to used the derived class' implementation of convertToFloat further down). So it is currently still a copy, but is a pointer copy rather than a Sensor object copy.

Reply:
Yeah that sounds good